ipaserver/install/service.py

```diff
@@ -208,10 +208,11 @@ def _ldap_mod(self, ldif, sub_dict=None, raise_on_err=True,
         args += ["-H", ldap_uri]
 
         if dm_password:
-            [pw_fd, pw_name] = tempfile.mkstemp()
-            os.write(pw_fd, dm_password)
-            os.close(pw_fd)
-            auth_parms = ["-x", "-D", "cn=Directory Manager", "-y", pw_name]
+            with tempfile.NamedTemporaryFile(
+                    mode='w', delete=False) as pw_file:
+                pw_name = pw_file.name
+                pw_file.write(dm_password)
+                auth_parms = ["-x", "-D", "cn=Directory Manager", "-y", pw_name]
```

Comment:
Please move the auth_param part out of the context manager. I got confused by it for a sec. It looked like you omitted ```pw_file.flush()``` like in other places. After a closer look it turns out the file name is used after the file has been closed.

```
with tempfile.NamedTemporaryFile(mode='w', delete=False) as pw_file:
    pw_file.write(dm_password)
pw_name = pw_file.name
auth_parms = ["-x", "-D", "cn=Directory Manager", "-y", pw_name]
```